test/03_USM.test.js

```diff
@@ -29,127 +43,280 @@ contract('USM', (accounts) => {
+      await timeMachine.revertToSnapshot(snapshotId)
+    })
+
+    describe("deployment", () => {
+      it("starts with correct fum price", async () => {
+        const fumBuyPrice = (await usm.fumPrice(sides.BUY))
+        // The FUM price should start off equal to $1, in ETH terms = 1 / price:
+        fumBuyPrice.toString().should.equal(wadDiv(WAD, priceWAD).toString())
+        const fumSellPrice = (await usm.fumPrice(sides.SELL))
+        fumSellPrice.toString().should.equal(fumBuyPrice.toString())
       })
     })
 
-    describe('minting and burning', () => {
+    describe("minting and burning", () => {
       it("doesn't allow minting USM before minting FUM", async () => {
-        await expectRevert(usm.mint(user1, user2, oneEth, { from: user1 }), 'Fund before minting')
+        await expectRevert(usm.mint(user1, user2, oneEth, { from: user1 }), "Fund before minting")
       })
 
-      it('allows minting FUM', async () => {
+      it("allows minting FUM", async () => {
         const fumBuyPrice = (await usm.fumPrice(sides.BUY))
         const fumSellPrice = (await usm.fumPrice(sides.SELL))
+        fumBuyPrice.toString().should.equal(fumSellPrice.toString()) // We should start with the buy & sell FUM prices equal
 
         await usm.fund(user1, user2, oneEth, { from: user1 })
-        const fumBalance = (await fum.balanceOf(user2))
-        fumBalance.toString().should.equal(oneEth.mul(priceWAD).div(WAD).toString()) // after funding we should have eth_price fum per eth passed in
+        const ethPool2 = (await weth.balanceOf(usm.address))
+        ethPool2.toString().should.equal(oneEth.toString())
+
+        const fumBalance2 = (await fum.balanceOf(user2))
+        // The very first call to fund() is special: the initial price of $1 is used for the full trade, no sliding prices.  So
+        // 1 ETH @ $250 should buy 250 FUM:
+        fumBalance2.toString().should.equal(wadMul(oneEth, priceWAD).toString())
+
+        const fundDefundAdj2 = (await usm.fundDefundAdjustment())
+        fundDefundAdj2.toString().should.equal(WAD.toString()) // First call to fund() doesn't move the fundDefundAdjustment
 
-        const newEthPool = (await weth.balanceOf(usm.address))
-        newEthPool.toString().should.equal(oneEth.toString())
+        const fumBuyPrice2 = (await usm.fumPrice(sides.BUY))
+        fumBuyPrice2.toString().should.equal(fumBuyPrice.toString()) // First fund() (w/o sliding prices) doesn't change FUM price
+        const fumSellPrice2 = (await usm.fumPrice(sides.SELL))
+        fumSellPrice2.toString().should.equal(fumSellPrice.toString())
 
-        const newFumBuyPrice = (await usm.fumPrice(sides.BUY))
-        newFumBuyPrice.toString().should.equal(fumBuyPrice.toString()) // Funding doesn't change the fum price
-        const newFumSellPrice = (await usm.fumPrice(sides.SELL))
-        newFumSellPrice.toString().should.equal(fumSellPrice.toString())
+        await usm.fund(user1, user2, oneEth, { from: user1 })
```

Comment:
Here, instead of funding `oneEth`, retrieve the `ethPool` value and fund by that. Then the test will be done on doubling the `ethPool`, whatever the initial `fund` was.

Let's try to parameterize the tests.